sycl/source/detail/device_impl.cpp

```diff
@@ -253,6 +256,58 @@ bool device_impl::has(aspect Aspect) const {
     return get_info<info::device::usm_restricted_shared_allocations>();
   case aspect::usm_system_allocator:
     return get_info<info::device::usm_system_allocator>();
+  case aspect::ext_intel_pci_address:
+    if (getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+            MDevice, PI_DEVICE_INFO_PCI_ADDRESS, sizeof(pi_device_type),
+            &device_type, &return_size) == PI_SUCCESS) {
+      return true;
+    } else {
+      return false;
+    }
+  case aspect::ext_intel_gpu_eu_count:
+    if (getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+            MDevice, PI_DEVICE_INFO_GPU_EU_COUNT, sizeof(pi_device_type),
+            &device_type, &return_size) == PI_SUCCESS) {
+      return true;
+    } else {
+      return false;
+    }
+  case aspect::ext_intel_gpu_eu_simd_width:
+    if (getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+            MDevice, PI_DEVICE_INFO_GPU_EU_SIMD_WIDTH, sizeof(pi_device_type),
+            &device_type, &return_size) == PI_SUCCESS) {
+      return true;
+    } else {
+      return false;
+    }
+  case aspect::ext_intel_gpu_slices:
+    if (getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+            MDevice, PI_DEVICE_INFO_GPU_SLICES, sizeof(pi_device_type),
+            &device_type, &return_size) == PI_SUCCESS) {
+      return true;
+    } else {
+      return false;
+    }
+  case aspect::ext_intel_gpu_subslices_per_slice:
+    if (getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+            MDevice, PI_DEVICE_INFO_GPU_SUBSLICES_PER_SLICE,
+            sizeof(pi_device_type), &device_type, &return_size) == PI_SUCCESS) {
+      return true;
+    } else {
+      return false;
+    }
+  case aspect::ext_intel_gpu_eu_count_per_subslice:
+    if (getPlugin().call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
+            MDevice, PI_DEVICE_INFO_GPU_EU_COUNT_PER_SUBSLICE,
+            sizeof(pi_device_type), &device_type, &return_size) == PI_SUCCESS) {
+      return true;
+    } else {
+      return false;
+    }
+  case aspect::ext_intel_max_mem_bandwidth:
+    // currently not supported
```

Comment:
can't it still be implemented with a call to a plugin already?

Reply:
zesMemoryGetBandwidth() is still a WIP.

Reply:
I mean you'd still query plugins here, and return PI_INVALID VALUE until better support is possible.

Reply:
May be done later too.